sqlglot/parser.py

```diff
@@ -1676,6 +1676,11 @@ def _parse_types(self):
         is_struct = type_token == TokenType.STRUCT
         expressions = None
 
+        if not nested and self._match_pair(TokenType.L_BRACKET, TokenType.R_BRACKET):
```

Comment:
what is this?

Reply:
is this if you do int[] which means int array?

Reply:
Yeah `int[]` means an int array.